File: api/lib/geo_json_formatter.rb
class GeoJSONFormatter
  def self.format(results, coords_key)
    results_with_latlong = results.select do |row|
      row[coords_key]
    end
    features = results_with_latlong.map do |row|
      properties = row.reject {|k| k == coords_key}
      {
        type: 'Feature',
        geometry: {
          type: 'Point',
          coordinates: row[coords_key].map {|c| c.to_f}
        },
        properties: properties
      }
    end

    {type: 'FeatureCollection',
     features: features}
  end
end

File: api-scala/app/lib/GeoJsonFormatter.scala
package lib

import play.api.libs.json._

object GeoJsonFormatter {

  def format(src: List[JsValue], key: String): JsObject = {

    val features = src map { objvalue =>
      val obj = objvalue.as[JsObject]
      val geoPoint = makeGeoPointDoubles(obj \ key)
      val props = obj - key

      Json.obj(
        "type" -> "Feature",
        "geometry" -> Json.obj(
          "type" -> "Point",
          "coordinates" -> geoPoint
        ),
        "properties" -> props
      )

    }

    Json.obj(
      "type" -> "FeatureCollection",
      "features" -> features
    )
  }

  private def makeGeoPointDoubles(pt: JsValue): JsValue = {
    val arrValues = pt.as[JsArray].value

    val doubleValues = arrValues.map {
      case JsString(s) => JsNumber(BigDecimal(s))
      case other => other
    }

    JsArray(doubleValues)
  }
}
